doc/feature/spikeReport.md

```diff
@@ -0,0 +1,338 @@
+         *    s <= t_s < r.getCurrentTime() (Note this could collapse to an
+         *    empty interval is r.getState() != 0)
+         *
+         * After successful f.wait_for or f.wait_until, the result is the same
+         * as above. If they time out, no observable state changes.
+         *
+         * If the state is FAILED or some other operation is still pending
+         * when read is called the result is undefined.
+         *
+         * @param min The minimum end time until which spikes will be read. If
+         *        UNDEFINED_TIMESTAMP, it will be considered as infinite.
+         */
+        boost::future< Spikes > read( float min );
+
+        /**
+         * Read all available spikes.
+         *
+         * This is a convenience function that behaves like read( -inf ) if
+         * getCurrentTime() is undefined and read( getCurrentTime( )) otherwise.
+         */
+        boost::future< Spikes > read();
+
+        /**
+         * Read spikes until getCurrentTime() == max, the end of the source is
+         * reached or the report closed.
+         *
+         * This function is very similar to the normal read, but instead of
+         * specifying a lower bound of getCurrentTime at return, it specifies
+         * a requested strict value. The preconditions and postconditions
+         * are the same except for those involving the parameter min which
+         * become:
+         *
+         * Precondition: max > getCurrentTime()
+         * Postcondition: If r.getState() == 0, then r.getCurrentTime() == max
+         *
+         * WIP
+         * - Why no min parameter, because it doesn´t make sense
+         * - hmmmm.. if read( 1000 ) times out, and then readUntil( 10 ) is
+         * called and succeeds, we have to move data from the container that
+         * was supposed to be returned in the first operation (and was kept
+         * for the next attempt) into the container for the second.x
+         */
+        boost::future< Spikes > readUntil( float max );
+
+        /**
+         * Forward to a given absolute timestamp.
+         *
+         * In read mode this means skipping data forward until the timestamp
+         * is made current. In write mode for streams, subscribers are notified
+         * about the progress.
+         *
+         * Preconditions:
+         * - toTimestamp >= getCurrentTime()
+         *
+         * Postconditions:
+         * Let:
+         *  - r be the SpikeReport
+         *  - f be the returned future by r.forward(toTimestamp)
+         * Then:
+         *  - After f.wait() returns r.getCurrentTime() == toTimestamp.
+         *  - The postconditions of read operations imply that this function
+         *    throws away the data previous to toTimestamp (or avoids reading
+         *    it at all if possible).
+         */
+        boost::future< void > forward( float toTimestamp );
+
+        enum StateFlags { ENDED = 1, FAILED = 2 };
```

Comment:
As I said, if this is a bit mask OK=0 doesn't make sense to me. If this is a State enum then I should be added. Probably it's better, the question is whether ENDED|FAILED is a state that makes sense.